activerecord/lib/active_record/encryption/configurable.rb

```diff
@@ -51,7 +51,11 @@ def encrypted_attribute_was_declared(klass, name) # :nodoc:
         def install_auto_filtered_parameters_hook(application) # :nodoc:
           ActiveRecord::Encryption.on_encrypted_attribute_declared do |klass, encrypted_attribute_name|
             filter_parameter = [("#{klass.model_name.element}" if klass.name), encrypted_attribute_name.to_s].compact.join(".")
-            application.config.filter_parameters << filter_parameter unless excluded_from_filter_parameters?(filter_parameter)
+
+            unless excluded_from_filter_parameters?(filter_parameter)
+              application.config.filter_parameters << filter_parameter
+              klass.filter_attributes += [encrypted_attribute_name.to_s]
```

Comment:
Personally I'm not concerned about this and I suggested concatenation, but I just wanted to mention that concatenation allocates new arrays so conceptually we could figure out a bit more efficient solution. However more efficient may be a bit more complex so personally I think it's not worth it and I'm ok with the concatenation

Just wanted to say it out loud if someone has strong concerns about arrays allocation